kotti/resources.py

```diff
@@ -111,10 +112,27 @@ def __getitem__(self, path):
             else:
                 return child
 
+        # Get a single child by its name
         if len(path) == 1:
+            # Add a cache dict if it doesn't exist yet
+            if '_child_cache' not in self.__dict__:
+                self._child_cache = {}
+            if path[0] in self._child_cache:
+                child = self._child_cache[path[0]]
+                # Only return an item from the cache if it's still a child
+                # (i.e. it hasn't been assigned to a different parent in the
+                # meantime).
+                if child.parent == self:
+                    return child
+                else:
+                    # remove item from the cache if the parent has changed.
+                    self._child_cache.pop(path[0])
```

Comment:
I left this one out of the superceding PR #449.  I might issue a new (and separate) PR when I'll be hitting this problem again and be able to describe the circumstances under which this would be a major benefit.  Sorry for the noise!